### PR TITLE
TimeBomb [recipe]

### PR DESCRIPTION
**Pull request by BachToTheFuture**

### CODE REVIEW

File: haiku-apps/timebomb/timebomb-0.4.0.recipe [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+DESCRIPTION="TimeBomb is a little customizable alarm clock application. \
+You can change its alarm sound, message, volume, and color!"
+HOMEPAGE="https://github.com/HaikuArchives/TimeBomb"
+COPYRIGHT="1999 Hailstone Software"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="09ce89ac6e3e028c4e88548b3e850e70bb392eb2"
+SOURCE_URI="https://github.com/HaikuArchives/timebomb/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="90393ffbcdc2a7be8ee173423e175af4e378c19eadf91371fd988e4f20ae98a6"
+SOURCE_FILENAME="2dPhysicsDemo-$portVersion-$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline haiku-apps/timebomb/timebomb-0.4.0.recipe]: Suggestion: :smile:
```diff
-SOURCE_FILENAME="2dPhysicsDemo-$portVersion-$srcGitRev.tar.gz"
+SOURCE_FILENAME="TimeBomb-$portVersion-$srcGitRev.tar.gz"
```

File: haiku-apps/timebomb/timebomb-0.4.0.recipe [hunk 2]
```diff
@@ -0,0 +1,46 @@
+SUMMARY="A loud digital alarm clock"
+DESCRIPTION="TimeBomb is a little customizable alarm clock application. \
+You can change its alarm sound, message, volume, and color!"
+HOMEPAGE="https://github.com/HaikuArchives/TimeBomb"
+COPYRIGHT="1999 Hailstone Software"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="09ce89ac6e3e028c4e88548b3e850e70bb392eb2"
+SOURCE_URI="https://github.com/HaikuArchives/timebomb/archive/$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline haiku-apps/timebomb/timebomb-0.4.0.recipe]: Suggestion:
```diff
+SOURCE_URI="https://github.com/HaikuArchives/timebomb/archive/$srcGitRev.tar.gz"
-SOURCE_URI="https://github.com/HaikuArchives/TimeBomb/archive/$srcGitRev.tar.gz"
```
(Not required, but slightly nicer.)

**Comment by bach5000** [comment]
@fbrosson Fixed!

File: haiku-apps/timebomb/timebomb-0.4.0.recipe [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+BUILD()
+{
+	make -C src $jobArgs OBJ_DIR=objects
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/TimeBomb
+	cp src/objects/TimeBomb $appsDir/TimeBomb
+	cp -r src/config.file $appsDir/TimeBomb
```
Comment (alaviss) [inline haiku-apps/timebomb/timebomb-0.4.0.recipe]: If you save the configuration file there, it's impossible to modify it. The correct way should be to make the app saves/load the configuration from `B_USER_SETTINGS_DIR`

**Review by alaviss — DISMISSED** [review]
You might also want to upstream that patchset :)

**Comment by bach5000** [comment]
Made a PR on upstream repo:
https://github.com/HaikuArchives/TimeBomb/pull/1

**Comment by alaviss** [comment]
@bach5000 Are you still working on this?

**Comment by bach5000** [comment]
@alaviss Currently I don't have Haiku available with me to experiment and fix things. It'll be a while until I get the computer with my Haiku back :(
I'll try to see what I can do.

**Comment by Begasus** [comment]
Pokes @bach5000  :)

**Comment by bach5000** [comment]
Will work on this once I am able to install Haiku on my computer. Sorry for the wait!

**Comment by bach5000** [comment]
Never mind. Could you make this a task for GCI 2019? I was unable to get things to work.

**Comment by Begasus** [comment]
Patch got merged upstream so I'm guessing there still needs work? I'll label it with "task" atm so GCI students can pick it up from here ...

**Comment by Begasus** [comment]
Crashes on launch.

**Comment by Begasus** [comment]
Switched srcGitRev to upstream commit, disabled for now as it crashes, please open issue/PR for further progress.

**Pull request merged**